stripe/src/main/java/com/stripe/android/model/PaymentMethodOptionsParams.kt

```diff
@@ -6,21 +6,37 @@ import kotlinx.android.parcel.Parcelize
 sealed class PaymentMethodOptionsParams(
     val type: PaymentMethod.Type
 ) : StripeParamsModel, Parcelable {
+    internal abstract fun createTypeParams(): List<Pair<String, Any?>>
+
+    override fun toParamMap(): Map<String, Any> {
+        val typeParams: Map<String, Any> = createTypeParams()
+            .fold(emptyMap()) { acc, (key, value) ->
+                acc.plus(
+                    value?.let { mapOf(key to it) }.orEmpty()
+                )
+            }
```

Comment:
can this just be
```suggestion
        val typeParams: Map<String, Any> = createTypeParams()
            .toMap().filterValues { it != null }
```
?

Reply:
Unfortunately Kotlin can't infer that the values will be non-null using this approach

Reply:
boo

Reply:
if you're not opposed to `!!`, we can just append a `.map { it!! }` and then kotlin is happy w/ it.